docs_src/design/adr/0005-Service-Self-Config.md

```diff
@@ -0,0 +1,74 @@
+- removal of the config-seed service (putting the current config-seed repositories in archive)
+- have each EdgeX micro service "self seed" - that is seed Consul with their own required configuration on bootstrap of the service.  Details of that bootstrapping process are below.
+
+### Operational parameters
+Operational parameters (op params for the rest of this document) are properties set by a service's command line arguements.  These include profile, configuration filename, configuration file directory, configuration provider URL, and overwrite flag.  The operational parameters are not set by any configuration (local file or Consul).  Op params include:
+
+- --configProvider or -cp (the configuration provider location URL)
+- --overwrite or -o (overwrite the configuration in the configuration provider)
+- --file or -f (the configuration filename)
+- --profile or -p (the profile name of configuration)
+- --confidir (the directory where the configuration file is found)
+- --registry or -r (use the registry)
+
+The distinction of op params versus configuration will be important later in this ADR.
+
+### Configuration Initializaiton
+Going forward, each service would have a local configuration file.  On intialization of the service (aka bootstrap of the service), the service would use the configuration to populate the configuration service (Consul) with data read from the configuration file.  In this way, each service self-seeds the configuration service.
+
+Details of the proposed process are as follows:
+- Service initializes and determines whether it should get its configuration from a local configuration file or from a configuration service by way of 
+  
+  a) the -cp / --configProvider op param
+
+  b) the edgex_configuration_provider environmental variable which overrides the op param above
+  
+- If the service is to use the local file, it reads the file associated with the profile, config filename and config file directory op params or environmental variables) and does not contact the configuration service (Consul) for any configuration information.
+- If the service is to use the configuration service, then the service will:
+  
+  1) call on Consul and check the root in Consul to see if it is already populated with configuration information provided by the service.  If so, the service will load and use the Consul provided configuration for that service.
+  2) If the root in Consul is not populated with configuration information provided by the service, the service will read the local configuration file and populate Consul with the configuration contents in that file.
+
+NOTE: as the services now self seed and the profile allows for the appropriate seeding to occur in the service itself, it will no longer be necessary to have a Docker configuration file in each of the service directories (example:  https://github.com/edgexfoundry/edgex-go/blob/master/cmd/core-data/res/docker/configuration.toml).  See Consequences below.
```

Comment:
Move this Note into the overrides section and add something about how env overrides can take the place of the minimal changes in the current `docker` profiles, i.e. `res/docker/configuration.toml`. Allowing the removal of 50% of the existing configuration.toml files. This approach is already be implemented and in use for app-service-configurable for Fuji release.

Reply:
The note you requested be moved doesn't feel like an override note to me.  So I left it.  Open to where it should be moved if you really think it is in the wrong place.  I did add a note on env overrides take place of changes in docker profiles.

Reply:
It is the env overrides per service that make this remove of docker profiles possible, thus why I feel is belongs in the overrides section.

How about a slight rewording of the note to help:
```
NOTE: as the services now self seed and deployment specific changes can be made via environment overrides, it will no longer be necessary to have a Docker configuration file in each of the service directories (example:  https://github.com/edgexfoundry/edgex-go/blob/master/cmd/core-data/res/docker/configuration.toml).  See Consequences below.
```

Reply:
Ohh, I see you add something under Overrides Notes.  So maybe just apply the note tweak above.

Reply:
tweaked